problem-solver/cxx/format-translators-module/agent/key-elements-orderer-agent/dummy_key_elements_orderer_agent.cpp

```diff
@@ -0,0 +1,81 @@
+#include "dummy_key_elements_orderer_agent.hpp"
+
+#include "keynodes/format_translators_keynodes.hpp"
+
+namespace formatTranslators
+{
+ScAddr DummyKeyElementsOrdererAgent::GetActionClass() const noexcept(false)
+{
+  return FormatTranslatorsKeynodes::action_order_key_elements;
+}
+
+bool DummyKeyElementsOrdererAgent::CheckInitiationCondition(ScActionInitiatedEvent const & event)
+{
+  return m_context.CheckConnector(GetActionClass(), event.GetOtherElement(), ScType::ConstPermPosArc);
+}
+
+ScResult DummyKeyElementsOrdererAgent::DoProgram(ScAction & action)
+{
+  try
+  {
+    ScAddr const & structure = action.GetArgument(1);
+    if (!m_context.IsElement(structure))
+      SC_THROW_EXCEPTION(utils::ExceptionInvalidParams, "Cannot find structure to order key elements");
+
+    ScAddrUnorderedSet elementsInOrder;
+    elementsInOrder.insert(structure);
+    auto const & it = m_context.CreateIterator5(
+        structure,
+        ScType::ConstPermPosArc,
+        ScType::Node,
+        ScType::ConstPermPosArc,
+        FormatTranslatorsKeynodes::rrel_main_key_sc_element);
+    if (it->Next())
+      elementsInOrder.insert(it->Get(2));
+
+    ScAddr lastConnector;
+    ScAddr const & elementsOrder = m_context.GenerateNode(ScType::ConstNodeTuple);
+    std::list nodeTypes = {
+        ScType::NodeClass, ScType::NodeLink, ScType::NodeRole, ScType::NodeNonRole, ScType::NodeTuple, ScType::Node};
+
+    for (auto const & nodeType : nodeTypes)
+      addNodesToOrder(structure, elementsOrder, elementsInOrder, lastConnector, nodeType);
+
+    ScAddr const & arc = m_context.GenerateConnector(ScType::ConstCommonArc, structure, elementsOrder);
+    m_context.GenerateConnector(ScType::ConstPermPosArc, FormatTranslatorsKeynodes::nrel_key_elements_order, arc);
+    action.SetResult(elementsOrder);
+    return action.FinishSuccessfully();
+  }
+  catch (utils::ScException const & exception)
+  {
+    m_logger.Error(exception.Description());
```

Comment:
```suggestion
    m_logger.Error(exception.Message());
```

Reply:
description is more informative
I don't why they both were added and I can add second log with message in it